### PR TITLE
[CDF-5654] Check Map[String, String] (a metadata field) in fromRow

### PR DESCRIPTION
**Pull request by exyi**
the `fromRow` macro now checks that `Map[String, String]` contains only strings, ignores null values and trims the values to the metadata size limit.

### CODE REVIEW

File: macro/src/main/scala/SparkSchemaHelper.scala [hunk 1]
```diff
@@ -60,12 +60,16 @@ class SparkSchemaHelperImpl(val c: Context) {
           param.typeSignature
         }
         val isSequenceWithOptionalVal = innerType <:< optionSeq
-        val isMapWithOptionalVal = innerType <:< optionMap
         val isSequenceOfAny = innerType <:< seqAny
-        val isMapOfAny = innerType <:< mapAny
+        val isMapOfString = innerType <:< mapString
+        if (!isMapOfString && innerType.getClass == classOf[Map[_, _]]) {
+          throw new Exception(s"Only Map[String, String] is supported, not $innerType")
```
Comment (wjoel) [inline macro/src/main/scala/SparkSchemaHelper.scala:66]: Isn't this excluding all other types of maps? It's possible to store a map<string, long> in a RAW column, for example, and that's allowed.

Comment (wjoel) [reply macro/src/main/scala/SparkSchemaHelper.scala:66]: Of course, this is for deriving the schema for our own types. :facepalm: All good then.

File: macro/src/main/scala/SparkSchemaHelper.scala [hunk 2]
```diff
@@ -74,38 +78,53 @@ class SparkSchemaHelperImpl(val c: Context) {
...
+             case x: Int => Some(x.toDouble)
+             case x: Float => Some(x.toDouble)
+             case x: Long => Some(x.toDouble)
              case x: BigDecimal => Some(x.toDouble)
              case x: BigInt => Some(x.toDouble)
              case _ => $throwError
            })"""
-          } else if (rowType <:< typeOf[Long]) {
+          } else if (rowType == typeOf[Float]) {
+            // do implicit conversion to double
```
Comment (wjoel) [inline macro/src/main/scala/SparkSchemaHelper.scala:101]: Either the comment or the code is wrong, should probably be `x.toDouble`?

Comment (exyi) [reply macro/src/main/scala/SparkSchemaHelper.scala:101]: Ah, sorry, the comment is lying ...

**Pull request merged**